lib/layers/dataUtils/facilities/methods.js

```
@@ -1,46 +1,209 @@
+        ":pk": { S: `facility::${orcs}` },
+        ":sk": { S: `${facilityType}::` },
+      },
+    };
+
+    if (Object.keys(filters).length > 0) {
+      queryObj = addFilters(queryObj, filters);
     }
+
+    const res = await runQuery(queryObj, limit, lastEvaluatedKey, paginated);
+    logger.info(`Facilities: ${res?.items?.length} found.`);
+    return res;
+  } catch (error) {
+    throw new Exception("Error getting facilities", {
+      code: 400,
+      error: error,
+    });
+  }
+}
+
+/**
+ * Retrieves facilities matching the specified Facility ID.
+ *
+ * @async
+ * @param {Object} orcs - The ORCS of the facility.
+ * @param {Object} facilityType - Type of facility to retrieve.
+ * @param {Object} facilityId - The ID of the facility to retrieve.
+ *
+ * @returns {Promise<Object>} Query response containing:
+ *   - items: facility object
+ *
+ * @throws {Exception} With code 400 if database operation fails
+ */
+async function getFacilityByFacilityId(orcs, facilityType, facilityId) {
+  logger.info("Get Facility By Facility Type and ID");
+  try {
+    let res = await getOne(
+      `facility::${orcs}`,
+      `${facilityType}::${facilityId}`
+    );
     return res;
   } catch (error) {
-    throw new Exception('Error getting facility', { code: 400, error: error });
+    throw new Exception("Error getting facility", { code: 400, error: error });
+  }
+}
+
+/**
+ * Validates that the sort key is valid from the facilityType and facilityId if provided.
+ *
+ * @param {Object} facilityType - Type of facility
+ * @param {Object} facilityId- The ID of the facility
+ * @param {Object} sk- The sort key of the facility
+ *
+ * @returns {void}
+ *
+ * @throws {Exception} With code 400 if there's a mismatch between sk and facilityType::facilityId
+ */
+function validateSortKey(facilityType, facilityId, sk) {
+  if (!sk && !facilityType && !facilityId) {
+    throw new Exception("sk or facilityType and facilityId are required.", {
```

should be `(!sk || (!facilityType && !facilityId))` to match this statement? Or are all 3 required?

---

The exception description was a bit of a misnomer I think so I've changed that in [f2daf02](https://github.com/bcgov/reserve-rec-api/pull/73/commits/f2daf028b15b2a31840be8c5c1d9fb7f2c224633), but it _does_ work properly in this format so I've kept it in. This check should throw an error if `sk`, `facilityType` and `facilityId` are missing, and if one of them exists it'll get to the next `if` statement where it compares their values.